src/main/groovy/org/craftercms/cli/commands/AddEnvironment.groovy

```diff
@@ -41,6 +46,15 @@ class AddEnvironment extends AbstractCommand {
                 config.password = auth.basic?.password
             }
             def client = HttpClient.getInstance(config)
+
+            if (auth.basic?.generateToken) {
+                config.token =  getTokenFromLogin(client)
+                client = HttpClient.resetClient(config)
+                config.token = CreateAccessToken.createAccessToken(client,
```

Comment:
Will we need to do some cleanup? What if we need to execute this several times? Will it reuse the same token?

Reply:
`add-environment` shouldn't be called several times, the idea is to set up your Studio once and then use the other CLI commands. You can have multiple Studios but each with a different environment name. 

There is a small bug in which if you call the `add-environment` again after a token is saved, it will create a new token, return the error of `Environment already exist`  but the token will be kept in Studio. I think this is not urgent to fix as the `add-environment` should be called only once.